cmd/oras/pull.go

```diff
@@ -135,7 +135,7 @@ func runPull(opts pullOptions) error {
 				}
 				// Skip s if s is unnamed and has no successors.
 				if len(ss) == 0 {
-					if _, loaded := printed.LoadOrStore(s.Digest.String(), true); !loaded {
+					if _, loaded := printed.LoadOrStore(s.Digest.String()+s.Annotations[ocispec.AnnotationTitle], true); !loaded {
```

Comment:
let's make a private function for this with `s` as input

Reply:
Added a function `generateKey(desc ocispec.Descriptor)`.